daemon/container.go

```diff
@@ -822,10 +843,15 @@ func (container *Container) monitorExec(execConfig *execConfig, callback execdri
 }
 
 func (c *Container) Attach(stdin io.ReadCloser, stdout io.Writer, stderr io.Writer) chan error {
-	return attach(&c.StreamConfig, c.Config.OpenStdin, c.Config.StdinOnce, c.Config.Tty, stdin, stdout, stderr)
+	err := attach(&c.StreamConfig, c.Config.OpenStdin, c.Config.StdinOnce, c.Config.Tty, stdin, stdout, stderr)
+	if err != nil {
+		return err
+	}
+	return nil
 }
 
 func (c *Container) AttachWithLogs(stdin io.ReadCloser, stdout, stderr io.Writer, logs, stream bool) error {
+	c.LogEvent("attach")
```

Comment:
I put it here because I think all flows go through here. If we're not streaming then we won't call c.Attach().